en/components/grid.md

```diff
@@ -22,345 +22,503 @@ _keywords: Ignite UI for Angular, UI controls, Angular widgets, web widgets, UI
 
-```typescript
-import { IgxGridComponent } from 'igniteui-angular/grid/grid.component';
-...
+    import { IgxGridComponent } from 'igniteui-angular/grid/';
+    // Or
+    import { IgxGridComponent } from 'igniteui-angular/main'
+    ...
 
-@ViewChild('myGrid', { read: IgxGridComponent }) public grid: IgxGridComponent;
-```
-
-<div class="divider--half"></div>
+    @ViewChild('myGrid', { read: IgxGridComponent })
+    public grid: IgxGridComponent;
 
 ### Usage
 
 Now that we have the grid module imported, let’s get started with a basic configuration of the **igx-grid** that binds to local data:
 
-```html
-<igx-grid #grid1 id="grid1" [data]="localData" [autoGenerate]="true"></igx-grid>
-```
+    <igx-grid #grid1 id="grid1" [data]="localData" [autoGenerate]="true"></igx-grid>
 
-The **id** property is a string value and is the unique identifier of the grid, while **data** binds the grid, in this case to local data.
+The **id** property is a string value and is the unique identifier of the grid which will be autogenerated if not provided, while **data** binds the grid, in this case to local data.
 
-The **autoGenerate** property tells the **igx-grid** to autogenerate columns based on the data source fields. Otherwise, the developer needs to explicitly define the columns and the mapping to the data source fields.
-
-<div class="divider--half"></div>
+The **autoGenerate** property tells the **igx-grid** to autogenerate columns based on the data source fields. It will also try to deduce the appropriate data type for the column if possible. Otherwise, the developer needs to explicitly define the columns and the mapping to the data source fields.
 
 ### Columns configuration
 
 **IgxGridColumnComponent** is used to define the grid's _columns_ collection and to enable features per column like **filtering**, **sorting**, and **paging**. Cell, header, and footer templates are also available.
 
 Let's turn the **autoGenerate** property off and define the columns collection in the markup:
 
-```html
-<igx-grid #grid1 [data]="data | async" [autoGenerate]="false" [paging]="true" [perPage]="6" (onColumnInit)="initColumns($event)"
-    (onCellSelection)="selectCell($event)">
-    <igx-column field="Name" [sortable]="true" header=" " [filtering]="true"></igx-column>
-    <igx-column field="AthleteNumber" [sortable]="true" header="Athlete number"></igx-column>
-    <igx-column field="TrackProgress" header="Track progress">
-        <ng-template igxCell let-col="column" let-ri="rowIndex" let-item="item">
-            <igx-linear-bar [striped]="false" [value]="item" [max]="100">
-            </igx-linear-bar>
+    <igx-grid #grid1 [data]="data | async" [autoGenerate]="false" [paging]="true" [perPage]="6" (onColumnInit)="initColumns($event)"
```

Comment:
Let's just assume I have the same comment for all code snippets that are not delimited.

Reply:
@ig-ealbert The code is indented as code, so it will still show as such.